spring-integration-amqp/src/main/java/org/springframework/integration/amqp/support/DefaultAmqpHeaderMapper.java

```diff
@@ -429,31 +429,31 @@ public static DefaultAmqpHeaderMapper outboundMapper() {
 	 * @return the default request headers for an inbound mapper.
 	 */
 	public static String[] inboundRequestHeaders() {
-		return safeInboundHeaders();
+		return new String[] { "*" };
```

Comment:
Not related to the changes, but I wonder why we don't see any deprecation warn for the factory methods based on the `@Deprecated` ctor:
```
public static DefaultAmqpHeaderMapper outboundMapper() {
	DefaultAmqpHeaderMapper mapper = new DefaultAmqpHeaderMapper();
	...
}
```
JDK bug?

We should provide `private` ctor on the matter and don't use deprecated API from our code.
Isn't it?